.github/eks-workflow-files/jio-eks/jio-jobset.yaml

```diff
@@ -0,0 +1,618 @@
+                echo "Test gateway connection"
+                python3 -c "
+                import os
+                import sys
+                import grpc
+
+                url = os.environ.get('GATEWAY_URL')
+                print(f'Gateway URL from env: {url}')
+
+                if not url:
+                    print('ERROR: GATEWAY_URL not set in environment')
+                    sys.exit(1)
+
+                try:
+                    print(f'Connecting to {url}...')
+                    channel = grpc.insecure_channel(url)
+                    grpc.channel_ready_future(channel).result(timeout=10)
+                    print(f'Successfully connected to gateway at {url}')
+                except Exception as e:
+                    print(f'Failed to connect to gateway: {e}')
+                    sys.exit(1)
+                "
+                # Start Ray Head
+
+
+                # 2. Start Ray Head
+                echo "Starting Ray Head..."
+                ray start \
+                  --head \
+                  --node-ip-address=${HOST_IP} \
+                  --port=${RAY_PORT} \
+                  --ray-client-server-port=${RAY_CLIENT_SERVER_PORT} \
+                  --num-cpus=48 \
+                  --num-gpus=8 \
+                  --block \
+                  --disable-usage-stats 2>&1 | tee ray-head.log &
+                PIDS+=($!)
+
+                sleep 10  # Let Ray start
+
+                # 3. Start vLLM Rollout Controller (no GPU, connects to Ray)
+                echo "Starting vLLM Rollout..."
+                cd /opt/jtbx/jax-inference-offloading/examples
+                export RAY_ADDRESS="${HOST_IP}:${RAY_PORT}"
+                CUDA_VISIBLE_DEVICES="" ray status
+                python -u rollout.py 2>&1 | tee rollout.log &
+                PIDS+=($!)
+
+                # Wait for all processes
+                echo "All vLLM components started, waiting..."
+                wait "${PIDS[@]}"
+                echo "vLLM node completed"
+              env:
+              - name: JOB_COMPLETION_INDEX
+                valueFrom:
+                  fieldRef:
+                    fieldPath: metadata.annotations['batch.kubernetes.io/job-completion-index']
+              - name: JOBSET_NAME
+                valueFrom:
+                  fieldRef:
+                    fieldPath: metadata.annotations['jobset.sigs.k8s.io/jobset-name']
+
+              # General config
+              - name: HF_TOKEN
+                valueFrom:
+                  secretKeyRef:
+                    name: huggingface-token
+                    key: token
+              - name: MODEL_NAME
+                value: "meta-llama/Llama-3.1-8B-Instruct"
+              - name: MODEL_PATH
+                value: ""  # or set to a path if using real weights
+                #NCCL
+              # - name: NCCL_DEBUG
+              #   value: "INFO"  # Change to WARN after debugging
+              - name: NCCL_PROTO
```

Comment:
NCCL's official guidelines are to avoid setting this variable explicitly whenever possible. Is this mandated by AWS?